src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

```diff
@@ -592,73 +594,99 @@ public String formatDateTime(GregorianCalendar time) {
 		return format.format(time.getTime());
 	}
 
-	private void ajustWidth() {
-		for (int i = 0; i < pColumwidth.length; i++) {
-			jTableInventory.getColumnModel().getColumn(i).setMinWidth(pColumwidth[i]);
-		}
-	}
-
 	private JComboBox<String> getComboBox() {
 		if (statusComboBox == null) {
 			statusComboBox = new JComboBox<String>();
-			statusComboBox.addItem("");
+			statusComboBox.addItem(MessageBundle.getMessage("angal.common.all.txt"));
 			List<String> statusList = medicalInventoryManager.getStatusList();
 			for (String status : statusList) {
 				statusComboBox.addItem(status);
 			}
 			statusComboBox.addActionListener(actionEvent -> {
-				InventoryBrowsingModel inventoryModel = new InventoryBrowsingModel();
-				totalRows = inventoryModel.getRowCount();
+				totalRows = medicalInventoryManager.getInventoryCount(InventoryType.main.toString());
 				startIndex = 0;
+				int page = startIndex / PAGE_SIZE + 1;
 				previous.setEnabled(false);
 				if (totalRows <= PAGE_SIZE) {
 					next.setEnabled(false);
 				} else {
 					next.setEnabled(true);
 				}
-				jTableInventory.setModel(new InventoryBrowsingModel(startIndex, PAGE_SIZE));
-				initialiseCombo(totalRows);
+				jTableInventory.setModel(new InventoryBrowsingModel(page, PAGE_SIZE));
+				initialisePagesCombo(totalRows);
 			});
 		}
 		return statusComboBox;
 	}
 
 	private JLabel getStatusLabel() {
 		if (statusLabel == null) {
-			statusLabel = new JLabel(MessageBundle.getMessage("angal.inventory.status.txt"));
+			statusLabel = new JLabel(MessageBundle.getMessage("angal.inventory.status.label"));
 			statusLabel.setHorizontalAlignment(SwingConstants.RIGHT);
 		}
 		return statusLabel;
 	}
 
-	public void initialiseCombo(int total_rows) {
+	public void initialisePagesCombo(int total_rows) {
 		int j = 0;
 
-		pagesCombo.removeAllItems();
-		for (int i = 0; i < total_rows / PAGE_SIZE; i++) {
-			j = i + 1;
-			pagesCombo.addItem(j);
-		}
-		if (j * PAGE_SIZE < total_rows) {
-			pagesCombo.addItem(j + 1);
-			under.setText("/" + (total_rows / PAGE_SIZE + 1 + " Pages"));
-		} else {
-			under.setText("/" + total_rows / PAGE_SIZE + " Pages");
+		// if totalRows = 0 we have at least 1 page
+		int totalPages = Math.max(1, (int) Math.ceil((double) totalRows / PAGE_SIZE));
+		for (int i = 1; i <= totalPages; i++) {
+			pagesComboBox.addItem(i);
 		}
+
+		ofPagesLabel.setText("/" + totalPages + " " + MessageBundle.getMessage("angal.common.pages.txt"));
```

Comment:
first single character strings can just use single quotes

second why not use a formatted message that will move the need for two concatenations

Something like:

angal.common.pages.fmt.txt                                                                                 = /{0} Pages